firmware/Core/Src/Logs.c

```diff
@@ -0,0 +1,36 @@
+#include "Logs.h"
+#include "esp8266.h"
+
+/***********************************************************************************************************************
+ * Code
+ **********************************************************************************************************************/
+
+void Logs_Init(void)
+{
+}
+
+int Logs_LogData(const LogData_t * LogData)
+{
+    if(Logs_WifiData(* LogData)):
+        return LOG_SUCCEEDED;
+    elif(Logs_FlashData(* LogData)):
+        return LOG_SUCCEEDED;
+    else:
+        return LOG_FAILED;
+}
+
+bool Logs_WifiData(const LogData_t * LogData)
+{
+  // check wifi connnection
+  // if wifi not connected, return 0
+  // Wifi_Init()
```

Comment:
Shouldn't Wifi_init be called from the Logs_Init fuction ?

Reply:
wifi_init starts the UART interrupt

Reply:
hmm

Reply:
What should generally happen is the Init function, which gets called once, starts the UART interrupt. Then the processor just waits until the event occurs. After it does, and it's done processing the interrupt, the last thing you do before you exit the interrupt service routine is get it to start the interrupt again.